lib/c-api/src/wasm_c_api/externals/function.rs

```diff
@@ -86,30 +94,36 @@ pub unsafe extern "C" fn wasm_func_new_with_env(
     let num_rets = func_sig.results().len();
     let inner_callback =
         move |env: &mut *mut c_void, args: &[Val]| -> Result<Vec<Val>, RuntimeError> {
-            let processed_args = args
+            let processed_args: wasm_val_vec_t = args
                 .into_iter()
                 .map(TryInto::try_into)
                 .collect::<Result<Vec<wasm_val_t>, _>>()
-                .expect("Argument conversion failed");
+                .expect("Argument conversion failed")
+                .into();
 
-            let mut results = vec![
+            let mut results: wasm_val_vec_t = vec![
                 wasm_val_t {
                     kind: wasm_valkind_enum::WASM_I64 as _,
                     of: wasm_val_inner { int64_t: 0 },
                 };
                 num_rets
-            ];
+            ]
+            .into();
 
-            let _traps = callback(*env, processed_args.as_ptr(), results.as_mut_ptr());
+            let _traps = callback(*env, &processed_args, &mut results);
```

Comment:
This might not be the same, we should be careful of the difference between fat pointers and thin pointers.

`as_ptr` and `as_mut_ptr` generally pass word sized pointers, references to slices are double words and references to vecs are different data entirely

Reply:
`processed_args` and `results` are now of kind `wasm_val_vec_t`. I believe it is safe to use `&` in this case.